### PR TITLE
Add tests for ranking evaluation with aliases

### LINKED ISSUE #29011: RankEval queries using aliases
**RankEval and aliases**:
Currently it is only possible to run rank_eval queries using an index name. In many situations we use aliases to hide the indices and it would be great to be able to use an alias name to run rank_eval queries to test search results.

### PR DESCRIPTION
**Pull request by cbuescher**
The ranking evaluation requests so far were not tested against aliases
but they should run regardless of the targeted index is a real index or
an alias. This change adds cases for this to the integration and rest
tests.

Closes #29011

### CODE REVIEW

File: modules/rank-eval/src/test/java/org/elasticsearch/index/rankeval/RankEvalRequestIT.java [hunk 1]
```diff
@@ -98,7 +107,9 @@ public void testPrecisionAtRequest() {
                 RankEvalAction.INSTANCE, new RankEvalRequest());
         builder.setRankEvalSpec(task);
 
-        RankEvalResponse response = client().execute(RankEvalAction.INSTANCE, builder.request().indices("test"))
+        // String indexToUse = randomBoolean() ? TEST_INDEX : INDEX_ALIAS;
```
Comment (mayya-sharipova) [inline modules/rank-eval/src/test/java/org/elasticsearch/index/rankeval/RankEvalRequestIT.java]: @cbuescher  With this commented out string, did you intend to use either `TEST_INDEX` or `INDEX_ALIAS`?

Comment (cbuescher) [reply modules/rank-eval/src/test/java/org/elasticsearch/index/rankeval/RankEvalRequestIT.java]: Yes, I think I was intending to randomize this and commented it out for some reason, will correct this.

**Comment by cbuescher** [comment]
@mayya-sharipova thanks for the review, I pushed another commit

**Review by mayya-sharipova — APPROVED** [review]
Thanks @cbuescher for the change. LGTM

**Pull request merged**